### PR TITLE
Specify main

### PR DESCRIPTION
**Pull request by Vanuan**
Main should be specified

### CODE REVIEW

**Comment by Vanuan** [comment]
There is a standard of packaging. If "main" is not specified, you'd have to put a full path: "npm_modules/pdfjs-dist/builds/pdfjs..."

**Comment by yurydelendik** [comment]
We would not like to promote usage of pdf.combined.js due to performance implications caused by the disabled worker. (See the explanation at https://github.com/erikras/react-pdfjs/issues/1)

**Comment by yurydelendik** [comment]
For node.js, I expect to introduce a UMD header at pdf.js level (currently blocked by #6683)

**Comment by Vanuan** [comment]
So, can we have 2 npm packages then? One for pdf client, other for pdf worker?

**Comment by yurydelendik** [comment]
What are definitions for "PDF client" and "PDF worker"?

**Comment by Vanuan** [comment]
You're saying pdf.combined.js is obsolete. So there are two files: pdf.js and pdf.worker.js
To use pdf.js with package managers like npm browserify, webpack (and others), we need the main entrypoint to be specified, so that "require('pdfjs')" and importScripts('pdfjs-worker') will both download respective files.
But since there are two files required, I believe there should be a way to use both files. The only option to do that is to split npm module into two modules: pdfjs and pdfjs-worker.

**Comment by yurydelendik** [comment]
Just setting "main" to "pdf.combined.js" will not do us any good, and looks like we need to address two problems here:
1. make require('package-name') work as expected in CommonJS
2. make packaging work for browserify, webpack, etc.

Here is the plan:
- [ ] #6683
- [ ] create "lib" folder and place cleaned ./src/{core,display,shared} files there in addition to build; set "main" to "lib/core/api.js"
- [ ] use AMD loader for development (such as RequireJS)
- [ ] add build tools and/or try to use the frameworks to allow custom packaging of the files in the lib folder (such as r.js, browserify, or webpack)

**Comment by Vanuan** [comment]
So you're saying that with introduction of UMD, there would be no need to specify  path to worker file?

File: make.js [hunk 1]
```diff
@@ -334,6 +334,7 @@ target.dist = function() {
       type: 'git',
       url: DIST_REPO_URL
     },
+    main: BUILD_DIR + 'pdf.combined.js',
```
Comment (yurydelendik) [inline make.js]: The 'pdf.js' shall be used here.

Comment (Vanuan) [inline make.js]: What about worker? Is worker optional?

**Comment by yurydelendik** [comment]
Closing in favor of #6785

**Pull request closed without merging**